### DISCUSSION TITLE
Where Can I start for HL7 v2 messaging (includes Medical Informations and documents) between two different systems?

### DISCUSSION THREAD

**Discussion by mustaqahmedara** [nextgenhealthcare/connect#4579] [Q&A]
I am new to HL7. After getting through several online resources and HL7 website, I got an brief idea how HL7 works. So now as an student intern, I would like to to develop a program wich communicate with other system to exchange data using HL7 standard.

To achive the above task, after going through several websites and getting knowledge of HL7 for past two weeks. I thought I am on the right track, but later I realized nope, unfortunately I am not in the right track. First I tried using Fhir implementation in visual studio 2019 using C#, but I misserably failed ending up only to create or update or delete the existing data from the Smart Server. I could not able to connect two different systems to exchange data like patient details or documets and so on.

Is there any other ways to accomplise my task using HL7 standard to exchange data between to different systems. !!!!!!!!!!!!!!!!!!!!!!!!!!!

I would like someone to guide me here.

#hl7-fhir
#hl7
#hl7-v2
#hl7-cda
#hl7-dotnetcore

**Comment by rjkroll** [comment]
It all depends on what type of data you want to send and your requirements.    Most patient data can be sent via normal Hl7 v.2x messages, which are the simplest to implement.  You could even base64 encode documents into the HL7 message or use Mirth to send them as attachments.  You could also send all of the patients data in one document, like a CCD-A, but that is more complex to actually accomplish.   And like you said, you could use FHIR (which has all of the stuff that a CCD-A has), but in order to use FHIR you need to setup a FHIR server, API's, database, etc.   And using FHIR turns all of your existing patient data to "resources", so in essense, you are just duplicating data into a new format.   FHIR seems to be the future, but if you do not have the requirements at this time to use it, then I would avoid it as its a big project to implement.

**Reply by mustaqahmedara** [reply]
Yes, that is why I tried to implement using FHIR. To be precise, I would like to use HL7 for transferring GDT (Device Data Transfer) data which is the German Standard for the exchange of data between medical information systems and medical devices in doctor's practices. 

Time is not an issue for me. 

So, there are software's available for doctors, in which they manage patient details. I would like to interoperability with the system through my system for the exchange of data. 

Okay even if I want to set up Fhir Server, API's, Database, etc how can I proceed!! Also, What will you suggest about Mirth Connect. I could not able understand clearly Mirth Connect and how to use it for my task. Where can I start to accomplish my task?

**Reply by rjkroll** [reply]
No worries!

The first thing you need to do is install Mirth.   Pretty simple, just download the right executable and install it.   There should be some simple directions to get you up and going on the Mirth download page.

Once you have Mirth installed, the good thing is that it can easily be transformed into a FHIR server/client by using the Mirth FHIR plugin.   Follow the directions and examples and user guides here : https://www.mirthcorp.com/community/wiki/pages/viewpage.action?pageId=38798954

Those will walk you through VERY simple examples of a FHIR Listener and FHIR sender, including some example channels that you can test with to see how everything works.   

Basically, Mirth can handle all of the receiving, responding, and sending of FHIR messages, including a full API set and the endpoints necessary for FHIR.    

FHIR, though, will require its own database since everything is stored as a 'resource'.   So, once you get the FHIR server up (using Mirth and the FHIR plugin) it will walk you through creating the sample/simple table and database necessary to begin using the channel.   Of course, its just a sample channel, with a sample table and you will need to extend upon the database schema to fit your needs.   Or, you could use something like FHIRBASE, which is a free, open-source database schema for FHIR.

Once that is done, you essentially have yourself a FHIR server.   The only thing left to do is get your existing patient data into the FHIR format and into the FHIR database.   This will require you to either do alot of manual mapping of your existing data and insert it as 'resources' into the FHIR database OR you can use a channel in Mirth that receives HL7 2.x messages(or CCD-A documents) parses them, converts them into FHIR, then imports them.   Luckily, there is a sample Mirth channel in that link above that does just that.   There are also various things API's on the web that are free that you can use to help you convert your data as well.


EDIT:  Below are some useful links that I used once I began my FHIR journey with Mirth:

https://dzone.com/articles/fhir-connector-implementation-using-mirth-hl7-stan
https://www.health-samurai.io/fhirbase
https://techno-soft.com/fhir-setup-on-mirth-connect.html/
http://hl7.org/fhir/
https://github.com/nextgenhealthcare/fhir-example-channels

**Reply by mustaqahmedara** [reply]
Thank you so much for your reply, It was good information. I will try to proceed by myself from here. It was great info for me.
